### PR TITLE
Server redux - .about.yml copier

### PR DESCRIPTION
**Pull request by jseppi**
Idea from @toolness in PR #13:

> This is a foolhardy attempt to convert team-api-server into a much simpler tool that just waits for GitHub webhook notifications from other 18F repos and copies their .about.yml's to _data/projects/<project-name>.yml in the team-api.18f.gov repo using the GitHub API.
> 
> This means:
>  * No more serving static files built from team-api.18f.gov. We'll move that over to a cloud.gov staticfile-buildpack, which will be pushed-to by a travis after_success hook on team-api.18f.gov.
>  * No more calling git from the command-line on the server-side. We'll use GitHub's API directly.
> 
> The above means that we remove ruby and git as dependencies, making this a pretty straightforward node app that can be deployed to cloud.gov via its node buildpack.

TODO:
- [x] more tests
- [x] test running in Travis
- [x] eshint linting in Code Climate
- [x] code review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,51 @@
...
-You can find the absolute path to `team-api` by running `which team-api`.
 
 ## Contributing
 
 1. Fork the repo (or just clone it if you're an 18F team member)
 2. Create your feature branch (`git checkout -b my-new-feature`)
-3. Make your changes and test them via `npm test` or `gulp test`
-4. Lint your changes with `gulp lint`
+3. Make your changes and test them via `npm test`
+4. Lint your changes
```
Comment (toolness) [inline README.md]: Should `npm test` also do linting?

File: index.js [hunk 2]
```diff
@@ -1,59 +1,58 @@
...
-        return reject(result);
-      }
-      console.log(operation + ' succeeded');
-      return resolve(result);
-    });
-  };
-}
+/**
+ * This is a small server that listens for push webhooks from GitHuh repos,
+ *  checks if a target YAML file (like `.about.yml`) has changed,
```
Comment (toolness) [inline index.js]: Is the extra space of indentation before the word `checks` here intentional? It's fine if it is, just wanted to make sure it wasn't a typo...

File: lib/GitHubFileCopier.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    this.githubOrg = options.githubOrg;
+    this.githubUser = options.githubUser;
+    this.githubAccessToken = options.githubAccessToken;
+    this.targetFile = options.targetFile;
+    this.destinationRepo = options.destinationRepo;
+    this.destinationPath = options.destinationPath;
+    this.destinationBranch = options.destinationBranch;
+
+    this.rp = requestPromise.defaults({
+      baseUrl: 'https://api.github.com/repos/',
```
Comment (toolness) [inline lib/GitHubFileCopier.js]: Nice, I didn't know they added `baseUrl` to request!! :+1:

File: README.md (hunk 1 above)
Comment (jseppi) [inline README.md]: I don't see that as a pattern generally. Also, I think we're using Code Climate for that kind of thing. I  think maybe I just set that up, but I haven't used Code Climate before: https://codeclimate.com/github/18F/team-api-server

File: lib/GitHubFileCopier.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      },
+    });
+  }
+
+  wasTargetUpdated(payload) {
+    if (!payload.ref) {
+      return false;
+    }
+
+    if (payload.repository.full_name.indexOf(this.githubOrg) !== 0 ||
```
Comment (toolness) [inline lib/GitHubFileCopier.js]: I wonder if it makes sense to do `indexOf(this.githubOrg + '/')` just to be safe/explicit?

File: package.json [hunk 5]
```diff
@@ -1,42 +1,44 @@
...
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/18F/team-api.git"
+    "url": "git+https://github.com/18F/team-api.18f.gov.git"
   },
   "author": "Mike Bland <michael.bland@gsa.gov>",
   "bugs": {
-    "url": "https://github.com/18F/team-api/issues"
+    "url": "https://github.com/18F/team-api.18f.gov/issues"
```
Comment (toolness) [inline package.json]: Should this and the URL above it actually point to `18F/team-api-server`?

Comment (jseppi) [inline package.json]: Yeah, totally

File: README.md (hunk 1 above)
Comment (toolness) [inline README.md]: Ok, I've seen it on some open-source projects like [p5.js](https://github.com/processing/p5.js), but those might be out of the ordinary... But if we don't do that, should we provide instructions on how to lint? When I first saw `6. Lint your changes` my first question was "how do I do that?"

**Comment by toolness** [comment]
This looks great to me!

File: README.md (hunk 1 above)
Comment (jseppi) [inline README.md]: Yeah, I'd actually like some guidance on what we want to do as a best practice for that. I use the `linter-eslint` plugin for atom.

Comment (toolness) [inline README.md]: Ah gotcha--maybe provide an easy "default" way to do it via the command-line, like `npm run lint`, but then have a separate section on more user-friendly ways of doing it depending on your editor?

**Comment by jseppi** [comment]
This project should probably be renamed to be more appropriately descriptive. Maybe like "team-about-aggregator" or something?

**Comment by toolness** [comment]
Good idea, though I think we might want to keep the prefix "team-api" just so one can see it's related to "team-api.18f.gov" just by the name... Maybe "team-api-webhook-server"? That way if we end up expanding the scope of this project for one reason or another, e.g by adding more webhooks that do stuff, we wouldn't have to rename it?

**Pull request merged**